SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LBDataStorage.cpp

```diff
@@ -65,17 +139,14 @@ FActorGroupStorage::FActorGroupStorage()
 	Components.Add(SpatialConstants::ACTOR_GROUP_MEMBER_COMPONENT_ID);
 }
 
-void FActorGroupStorage::OnAdded(Worker_EntityId EntityId, const SpatialGDK::EntityViewElement& Element)
+void FActorGroupStorage::OnComponentAdded(Worker_EntityId EntityId, Worker_ComponentId ComponentId, Schema_ComponentData* Data)
 {
-	for (const auto& Component : Element.Components)
+	if (ensureAlways(SpatialConstants::ACTOR_GROUP_MEMBER_COMPONENT_ID == ComponentId))
 	{
-		if (SpatialConstants::ACTOR_GROUP_MEMBER_COMPONENT_ID == Component.GetComponentId())
-		{
-			Schema_Object* GroupObject = Schema_GetComponentDataFields(Component.GetUnderlying());
-			int32 GroupId = Schema_GetUint32(GroupObject, SpatialConstants::ACTOR_GROUP_MEMBER_COMPONENT_ACTOR_GROUP_ID);
-			Groups.Add(EntityId, GroupId);
-			Modified.Add(EntityId);
-		}
+		Schema_Object* GroupObject = Schema_GetComponentDataFields(Data);
+		int32 GroupId = Schema_GetUint32(GroupObject, SpatialConstants::ACTOR_GROUP_MEMBER_COMPONENT_ACTOR_GROUP_ID);
```

Comment:
Although again, I'd just use the entire `ActorGroupMember`.
```suggestion
		ActorGroupMember GroupMemberData;
		GroupMemberData.ApplySchema(GroupObject);
		int32 GroupId = GroupMemberData.ActorGroupId;
```

Reply:
Could do, although at this point I'll just replace it with the template version.